### ISSUE TITLE
Allow fully configurable URIs for Status and Credentials

### ISSUE THREAD

**Issue by decentralgabe** [TBD54566975/ssi-service#640]
At present, using the `api_host` we can set a per-service value to be stamped in all VCs and Status Lists the service produces. We can take this a step further, by allowing per-service configuration of this value...and not just for the base host, but for the entire path.

- An implementer should be able to set any path to externally access VCs/Status Lists.
- An implementer may choose to not have credentials be externally resolvable (this should be the default - for privacy reasons)
- An implementer may choose to have a different URI for hosting a status list separate to the service itself

**Comment by michaelneale** [comment]
@decentralgabe is this a #goodfirstissue type of thing for contributors potentially?

**Comment by michaelneale** [comment]
relevant line of code: https://github.com/TBD54566975/ssi-service/blob/main/pkg/service/credential/status.go#L72 

```go
statusListURI := fmt.Sprintf("%s/status/%s", config.GetServicePath(framework.Credential), statusListID)
```

(just so I remember it)

**Comment by michaelneale** [comment]
@decentralgabe so am I right in thinking if fully specified statusListURI should be a constant url for the whole status list?

**Comment by michaelneale** [comment]
I think a common case would be a service in front of ssi which would choose to expose the statuslist (and only the status list). Could the api_host be used for this or a separate config?

**Comment by decentralgabe** [comment]
> @decentralgabe so am I right in thinking if fully specified statusListURI should be a constant url for the whole status list?

yes though I imagine what we want is the ability to set a base path and then let the service handle the ID generation

so you would configure `statusBaseURL` to `https://my-website.com/status` and then the service would add `https://my-website.com/status/{status-list-id}`

**Comment by michaelneale** [comment]
yep I think you are right - given that is the preferred pattern, and makes it clearer what they are for. statusBaseURL could perhaps be optionally a full path in which case it won't append the ID (but that means it is up to implementers to ensure status less is updated elsewhere, so maybe that is a bit premature and YAGNI for now)

**Comment by decentralgabe** [comment]
how would it work without the ID given that there are many status lists? for each <schema, issuer> pair?

**Comment by decentralgabe** [comment]
fixed in #660

**Comment by michaelneale** [comment]
thanks @decentralgabe
